labs/b2.md

```diff
@@ -0,0 +1,145 @@
+
+## A quick intro to `vim`
+### Why `vim`?
+ - It's a descendant of vi, which was written in Berkeley by Bill Joy, who went on to found Sun Microsystems.
+ - Sometimes you will be suddenly thrown into `vim` via merging git conflicts or other programs.
+ - It's included in practically every UNIX environment.
+ - You can be very productive when familiar with it.
+
+### Hello World
+     Vim is a modal text editor, meaning that you can change editing modes in order to do different things. There are 3 primarily used modes: **Normal**, **Insert**, and **Visual** mode.
+#### Normal mode:
+ - Used for moving around and editing text
+   - `hjkl` to move left, up, down, and right
+   - `G` to move to end of file, `gg` to move to beginning
+   - `i` to enter **i**nsert mode (`a`, `o` also change mode in different ways)
+   - `dd` to cut a line
+   - `yy` to copy a line
+   - `p` to paste
+   - `/` to search
+   - `u` to undo
+ - Type in commands with `:`
+   - Save with `:w`
+   - Exit with `:q`
+ - Explore more commands online!
+
+#### Insert mode:
+ ![Example](https://d26aqo05ggejx9.cloudfront.net/i.gif)
+ - Used for editing text like a usual editor
+ - Arrow keys to move
+ - Esc to exit to normal mode (lots of people bind it to Caps Lock)
+
+#### Visual mode:
+ - Enter with `v` from normal mode
+ - Used to select text visually
+ - Modify selection with normal mode movement commands
+ - Use `o` to move the cursor to the other side of the selection
+ - Yanking, deleting, and pasting use `y`, `d`, `p` (sound familiar?)
+
+  A key feature of `vim` is **chaining together commands**.
+  Normal mode is essentially a massive amount of shortcuts that you can combine to quickly navigate and edit a file.
+  Want to move down 3 lines?
+  You know that `j` means move down 1 line, so you can use `3j` to move down 3.
+  `d` is for deletion and `w` is to jump to the next word, so what does `dw` do?
+
+### Questions
+ Try playing around with [lab2.md](https://github.com/tanx16/decal-web/blob/master/labs/b2.md) while looking up some new commands. Use `wget` to download it!
+ - How would you delete the previous 10 lines?
+ - How would you jump back to the shell without exiting `vim`?
+ - How would you edit a new file alongside another file?
+ - How would you indent a block of text?
+
+### If you're interested in `emacs` instead
+ - Use [this](http://readline.kablamo.org/emacs.html) cheatsheet
+
+## A quick intro to tmux
+ ![Example](https://lightrains.com/assets/img/blog/tmux-tips.png)
+### Why tmux?
+ - You can open multiple windows when sshed into a machine.
+ - You can go compile and run programs while editing them.
+ - You can logout and ssh back in without having to reopen all your files.
+
+### Getting Started:
+ - Start a session with `tmux`.
+ - Detach from a session with `Ctrl-b d` (press d after releasing `Ctrl-b`)
+ - Split into 2 panes with `Ctrl-b %` (vertical) and `Ctrl-b "` (horizontal)
+ - Swap current pane with `Ctrl-b o`
```

Comment:
Maybe add some stuff about adding more windows with `Ctrl-b c` and switching between windows? Not too critical, but I find multiple windows pretty useful when you have opened too many panes.